File: test_socket_connection.js
const { io } = require('socket.io-client');

async function testSocketConnection() {
  console.log('🧪 Testing Socket.IO connection to port 5001...');
  
  try {
    // Connect to the Socket.IO server
    const socket = io('http://localhost:5001', {
      transports: ['websocket', 'polling'],
      timeout: 5000
    });

    // Set up event listeners
    socket.on('connect', () => {
      console.log('✅ Connected to Socket.IO server!');
      console.log(`Socket ID: ${socket.id}`);
      console.log(`Transport: ${socket.io.engine.transport.name}`);
    });

    socket.on('disconnect', (reason) => {
      console.log(`❌ Disconnected: ${reason}`);
    });

    socket.on('error', (error) => {
      console.log(`❌ Socket error: ${error}`);
    });

    socket.on('new_pool', (data) => {
      console.log('📡 Received new_pool event:', data);
    });

    socket.on('health', (data) => {
      console.log('📡 Received health event:', data);
    });

    socket.on('pool_status_6', (data) => {
      console.log('📡 Received pool_status_6 event:', data);
    });

    // Wait for connection or timeout
    await new Promise((resolve, reject) => {
      const timeout = setTimeout(() => {
        reject(new Error('Connection timeout after 10 seconds'));
      }, 10000);

      socket.on('connect', () => {
        clearTimeout(timeout);
        resolve();
      });

      socket.on('connect_error', (error) => {
        clearTimeout(timeout);
        reject(error);
      });
    });

    console.log('✅ Socket connection test successful!');
    console.log('📡 Listening for events... (press Ctrl+C to exit)');
    
    // Keep the connection alive
    process.on('SIGINT', () => {
      console.log('\n🛑 Disconnecting...');
      socket.disconnect();
      process.exit(0);
    });

  } catch (error) {
    console.error('❌ Socket connection test failed:', error.message);
    process.exit(1);
  }
}

testSocketConnection(); 

File: test_socket_connection.py
#!/usr/bin/env python3

import asyncio
import socketio
import json
from datetime import datetime

# Create a Socket.IO client
sio = socketio.AsyncClient(
    logger=True,
    engineio_logger=True
)

@sio.event
async def connect():
    print(f"✅ Connected to server with ID: {sio.sid}")
    
    # Send a test message
    await sio.emit('test_connection', {
        'client_id': sio.sid,
        'timestamp': datetime.now().isoformat(),
        'message': 'Test from simple client'
    })
    print("📤 Sent test message")

@sio.event
async def disconnect():
    print("❌ Disconnected from server")

@sio.event
async def test_response(data):
    print(f"✅ Test response received: {json.dumps(data, indent=2)}")

@sio.event
async def health(data):
    print(f"🏥 Health check received: {json.dumps(data, indent=2)}")

@sio.event
async def pool_status_6(data):
    print(f"🚀 Pool status 6 received: {json.dumps(data, indent=2)}")

@sio.event
async def arbitrage_opportunity(data):
    print(f"🎯 Arbitrage opportunity received: {json.dumps(data, indent=2)}")

@sio.event
async def paper_trading_update(data):
    print(f"💰 Paper trading update received: {json.dumps(data, indent=2)}")

@sio.event
async def message(data):
    print(f"📨 Generic message received: {json.dumps(data, indent=2)}")

async def main():
    try:
        print("🔌 Connecting to Socket.IO server...")
        await sio.connect('http://localhost:5001')
        
        print("⏳ Waiting for events...")
        await asyncio.sleep(30)  # Wait 30 seconds for events
        
    except Exception as e:
        print(f"❌ Error: {e}")
    finally:
        await sio.disconnect()

if __name__ == '__main__':
    asyncio.run(main()) 